### DISCUSSION TITLE
Issues with Update 20210330

### DISCUSSION THREAD

**Discussion by sibezwo** [rern/rAudio#283] [Bug Reports]
Hi Rern

After the last update I noticed the following:

- Access Point was switched on automatically

- If I turn it off, it will turn on again after a reboot

- Boot duration for userspace now almost twice as long (34s)

- Play on Startup no longer works

My RPi 3B is connected to a wireless router.

Kind regards,

Thomas

**Comment by Leobo71** [comment]
I have the same problems. When restarting, hotspot switches on automatically. Sound shakes on my jbl bluetooth speaker.
Greetings

**Reply by Leobo71** [reply]
Sorry I had wrong config.txt. Everything works fine after the latest update.

**Comment by sibezwo** [comment]
Something went wrong after the 20210402 update. rAudio was no longer reachable, not even via the access point. I had to rewrite the SD card. Now everything is fine again. Thanks for the help.

Thomas

**Reply by Leobo71** [reply]
In config.txt I had wifi0 and wifi files, each mall connected to the access point and not to my WiFi.

**Comment by somax80** [comment]
I've the same problem on wifi.
My **Raspberry pi3b** with **20210403** release start without active wifi connection, then I activate the connection and it connects correctly to router.
It seems ok, but when i reboot the system without network cable, it start with wifi configured as an access point.
On my network I've haven't experienced any "long time" to connect on my router. I've already try to increase the number of times it try to connect in startup.sh:
for i in {1..10}; do (line 97)
with no success.
With the 20210315 version that i've previously installed i haven't any type of  this problem.

So, merging **20210403** with **20210328** release,  i've manually copied this two files:
/srv/http/bash/startup.sh
/srv/http/bash/networks.sh
from **20210328** release into my **rAudio 20210403 installation**, and the problem on wifi startup magically disappeared. 
All is ok, the wifi start and connect correctly during startup; if i reboot it takes wifi connection (with dynamic IP) immediately.

Maybe is there some problem on release 20210330 that "Improve - Wi-Fi connection on startup"?
I've download latest release 20210414, but seems that these files are the same as 20210330, so i think the problem is still present.

Thanks
Max

**Reply by rern** [reply]
Hi Max,

With the 2 files replaced by 20210328 ones, please try:
- Disable Wi-Fi roaming:
    - `systemctl disable netctl-auto@wlan0`
    - Reboot

- If Wi-Fi failed:
    - `systemctl enable netctl-auto@wlan0`
    - Reboot

- Get revised `startup.sh`:
    ```sh
    mv /srv/http/bash/startup.sh{,.backup}
    wget https://github.com/rern/rAudio-1/raw/UPDATE/srv/http/bash/startup.sh -O /srv/http/bash/startup.sh
    ```
    - Reboot

- If Wi-Fi failed:
    - `mv /srv/http/bash/startup.sh{.backup,}`
    - Reboot

**Reply by somax80** [reply]
I follow all the steps:
disable wifi-roaming, reboot and then **wifi failed** to start on boot.
So re-enable wifi roaming, reboot, and then **wifi is ok**, started correctly on boot.
 
I backed up startup.sh and get latest version from git, then reboot.
**Now magically Wifi doesn't fail, started correctly at boot**
_wlan0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.1.125  netmask 255.255.255.0  broadcast 192.168.1.255
        ..._

So now could I update also networks.sh script from git?
And so now could I get latest version from web interface (from addons menu) and install release 20210414? 
Or i'll experience che same problem on wifi startup?

thanks,
  Max

**Reply by somax80** [reply]
I've updated from web interface (addons) to latest release 20210414.
After that i've got the same problem on wifi during startup (wifi disabled).
So I copy startup.sh and networks.sh from 20210328 release to my rAudio-1 installation and then i follow the steps described by rern (disable wifi roaming, reboot, enable wifi roaming, reboot, backing up startup.sh (old) and get latest startup.sh from git, reboot).
Now i've the latest release (except networks.sh) and the wifi goes up&running correctly on boot.

regards,
  Max

**Comment by R101G** [comment]
I suspect this is the same problem I reported as #290. When I install the wifi dongle driver I enable netctl-auto@wlan0 as part of the process. If an update disabled it, that could explain why a new install fixed the problem for me - it had nothing to do with the new image.

**Reply by somax80** [reply]
Just a detail that i haven't reported: i don't have any wifi dongle, i use raspberry pi 3b v1.2 with integrated wifi.

**Reply by rern** [reply]
Max,
@R101-Github,

Please try `UPDATE` branch.
 
Max:
- Addons page > rAudio 1 - long-press `Update` button
- OK for `UPDATE` branch
- Verify the connection is `enabled`: `netctl is-enabled "SSID"` (replace "SSID" with yours)
- Reboot

**Reply by R101G** [reply]
@rern 
Reconnected after UPDATE and reboot. Thanks.

**Reply by rern** [reply]
The fix has been included in today's update.

**Reply by somax80** [reply]
With update branch i was able to have wifi up&running after reboot.
Many Thanks.
Max
